jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/solver/SolverRX.java

```diff
@@ -94,7 +94,7 @@
         Iterator<Quad> dsgIter = accessData(patternTuple, nodeTupleTable, anyGraph, filter, execCxt);
 
         Binding input = bnid.isEmpty() ? BindingFactory.empty() : new BindingTDB(bnid, nodeTable);
-        Iterator<Binding> matched = Iter.iter(dsgIter).map(dQuad->RX.matchQuad(input, dQuad, tGraphNode, tPattern)).removeNulls();
+        Iterator<Binding> matched = Iter.iter(dsgIter).map(dQuad->SolverRX4.matchQuad(input, dQuad, tGraphNode, tPattern)).removeNulls();
```

Comment:
Does using the general purpose solver mean `Node` get materialized sooner or is the TDB `DatasetGraph` going to return special `Node` instances backed by internal Node IDs?

Reply:
For non-RDF-star, patterns, it returns instances backed by internal Node IDs (`BindingNodeId`, `BindingTDB`).

Inside `<<:s ?p ?o>>`, things have to switch to Nodes because the NodeId indexes do not store embedded triples presently.

If (1) there is any performance impact and (2) people are using RDF-star enough, this can change later - it is a on-disk file format change. The current `SolverRX` do not require an on-disk change and so no data reload.